File: src/main/java/com/ptmanager/backend/auth/AuthController.java
package com.ptmanager.backend.auth;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.ptmanager.backend.domain.User;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/login")
    public LoginResponse login(@Valid @RequestBody LoginRequest request) {
        User user = authService.login(request.email(), request.password());
        return new LoginResponse("dev-token-" + user.getId(), user);
    }

    public record LoginRequest(
        @Email @NotBlank String email,
        @NotBlank String password
    ) {
    }

    public record LoginResponse(
        String accessToken,
        User user
    ) {
    }
}

File: src/main/kotlin/com/ptmanager/backend/auth/AuthController.kt
package com.ptmanager.backend.auth

import com.ptmanager.backend.auth.dto.LoginRequest
import com.ptmanager.backend.auth.dto.LogoutRequest
import com.ptmanager.backend.auth.dto.RefreshRequest
import com.ptmanager.backend.auth.dto.SignupRequest
import com.ptmanager.backend.auth.dto.TokenResponse
import com.ptmanager.backend.domain.User
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val authService: AuthService,
) {

    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    fun signup(@Valid @RequestBody request: SignupRequest): TokenResponse =
        authService.signup(request.email, request.password, request.name, request.role)

    @PostMapping("/login")
    fun login(@Valid @RequestBody request: LoginRequest): TokenResponse =
        authService.login(request.email, request.password)

    @PostMapping("/refresh")
    fun refresh(@Valid @RequestBody request: RefreshRequest): TokenResponse =
        authService.refresh(request.refreshToken)

    @PostMapping("/logout")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun logout(
        @AuthenticationPrincipal userId: Long,
        @RequestBody(required = false) request: LogoutRequest?,
    ) {
        authService.logout(userId, request?.deviceToken)
    }

    @GetMapping("/me")
    fun me(@AuthenticationPrincipal userId: Long): User = authService.getMe(userId)
}
